### ISSUE TITLE
Why execute from right to left?

### ISSUE THREAD

**Issue by FeldrinH** [uiua-lang/uiua#2]
The choice to execute code from right to left feels really weird to me. Mainly because of two reasons:

1. Coming from writing systems where text is written left to right It just feels a lot more natural to read in that direction, but reading Uiua programs from left to right makes them very difficult to understand.

2. Because Uiua code is executed right to left It seems to me that the natural way to write Uiua code is also right to left.
However, every time you type a symbol your cursor is placed to the right of that symbol. This means that if you want to type multiple symbols you have to constantly move your cursor to the left, over the newly added symbols. This feels really clunky.

I'm curious about the rationale of this decision and if there is any possiblity that it might be reevaluated at some point in the future.

PS: I will admit that I'm not a very familiar with stack based languages so maybe there is some crucial thing here that I just don't get.

**Comment by kaikalii** [comment]
There are two primary reasons I made the language go right-to-left:
1. Most array languages are read right-to-left. Do they have the same problem with the cursor? I'm not familiar.
2. I've always found the reverse-polish-notation of most stack languages kind of weird. It contrasts with Lisp or ML derivatives where the function goes *before* its operands. Even Algol derivatives mostly do this. Things like `1 2 +` look so strange to me. Another nice effect of this is that the "big" operations can be at the beginning.

That being said, I *am* sympathetic to counter-arguments, mainly for the cursor reason you cited.
It gets even worse with multi-line functions. They are parsed as if they are on a single line, so code before and after the `()`s still runs "right-to-left", even if the code on the "right" is on line further down.
For examples of what I mean, see the [HTTP server example](https://github.com/uiua-lang/uiua/blob/main/examples/http_server.ua).

In addition, splitting code into multiple lines is weird with the current right-to-leftness. You have to split the line with a newline, then swap it with the line above.

One potential solution that would work without actually changing the language would be to change the way the cursor moves automatically in the online editor so it jumps to the beginning of your current glyph. However, I think this would be kind of weird and counter-intuitive, and wouldn't extend to other editors.

It may be worth it to make a branch with a modified version of the parser just to see how it feels to write. It actually wouldn't be that big a change to the interpreter code itself.
What *would* be a good chunk of work is reversing all the existing examples and tests.

**Comment by kaikalii** [comment]
Just to try it out, I created a new branch called `left-to-right`.
I haven't changed any examples or tests yet, but basic code seems to work.
I actually didn't change the parser at all, I just took out calls to `.rev()` on some iterators in the compiler 😆.

Writing/reading it is kind of weird for me. It's like looking at a picture of yourself where you aren't flipped like you are in a mirror. Not wrong just...odd.
